security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -622,30 +622,40 @@ func (sc *SecretManagerClient) handleFileWatch() {
 			if !(isWrite(event) || isRemove(event) || isCreate(event)) {
 				continue
 			}
-			cacheLog.Debugf("file cert update: %v", event)
-
-			sc.certMutex.RLock()
-			resources := sc.fileCerts
-			sc.certMutex.RUnlock()
-			// Trigger callbacks for all resources referencing this file. This is practically always
-			// a single resource.
-			for k := range resources {
-				if k.Filename == event.Name {
-					sc.CallUpdateCallback(k.ResourceName)
+			// Typically inotify notifies about file change after the event i.e. write is complete. It only
+			// does some housekeeping tasks after the event is generated. However in some cases, multiple events
+			// are triggered in quick succession - to handle that case we debounce here.
+			go debounce(100*time.Millisecond, func() {
```

Comment:
I don't see how this debounces - won't we still process all of the events, they will just be delayed by 100ms?

Reply:
hmm.. Sorry. I pushed the wrong commit. PTAL now.

Reply:
Thinking about this more, we may not be able to debounce here properly because we are using the `event.Name` to trigger call back and if we debounce and ignore that event, we may skip sending update. 

So may be leave it as is?

Reply:
Ok. I changed it to capture events and trigger. See if this is better